ddtrace/internal/telemetry/writer.py

```diff
@@ -116,10 +117,16 @@ def periodic(self):
                     exc_info=True,
                 )
 
-    def shutdown(self):
-        # type: () -> None
-        self._app_closing_event()
-        self.periodic()
+    def on_shutdown(self):
+        if self._enabled:
+            self._app_closing_event()
+            self.periodic()
+
+    def _stop_service(self, *args, **kwargs):
+        # type: (...) -> None
+        # FIXME: don't join() on stop(), let the caller handle this x2
```

Comment:
@brettlangdon  Should I make this fix here, make this fix in a follow up PR or is this a pattern we plan to keep

Reply:
I am not sure I know enough context behind this to know what needs to be fixed, maybe @Kyle-Verhoog has the context?

Reply:
This comment copied was from `AgentWriter._stop_service`. I believe it was added by Matt but this issue was never addressed. I'll remove the comment since this is a pattern that has existed for years now.